File: files/300-399/392/CH-392.py
import pandas as pd

path = "300-399/392/CH-392 Table Transformation.xlsx"
input = pd.read_excel(path, usecols="B:C", nrows=25, skiprows=2)
test = pd.read_excel(path, usecols="E:H", nrows=6, skiprows=2)

input["group"] = (input["Col1"] == "Date").cumsum()
result = input.pivot(index="group", columns="Col1", values="Col 2").reset_index(drop=True)
result = result[["Date", "Customer", "Product", "Sale"]]

print((result == test).all().all())
# Output: True

File: files/300-399/392/CH-392.R
library(tidyverse)
library(readxl)

path <- "300-399/392/CH-392 Table Transformation.xlsx"
input <- read_excel(path, range = "B3:C27")
test <- read_excel(path, range = "E3:H9")

result = input %>%
  mutate(group = cumsum(Col1 == "Date")) %>%
  pivot_wider(names_from = Col1, values_from = `Col 2`) %>%
  select(-group) %>%
  mutate(Sale = as.numeric(Sale))

all.equal(result, test)
## [1] TRUE
